uplink/config.go

```diff
@@ -41,10 +43,84 @@ type RSConfig struct {
 // EncryptionConfig is a configuration struct that keeps details about
 // encrypting segments
 type EncryptionConfig struct {
-	Key       string      `help:"root key for encrypting the data"`
-	BlockSize memory.Size `help:"size (in bytes) of encrypted blocks" default:"1KiB"`
-	DataType  int         `help:"Type of encryption to use for content and metadata (1=AES-GCM, 2=SecretBox)" default:"1"`
-	PathType  int         `help:"Type of encryption to use for paths (0=Unencrypted, 1=AES-GCM, 2=SecretBox)" default:"1"`
+	// TODO: if/v3-1541 read TODO in Key method why the following field cannot
+	// be added
+	// RootKey     *storj.Key  // The root key for encrypting the data and read it from KeyFilepath, see Key method.
+	KeyFilepath string      `help:"the path to the file which contains the root key for encrypting the data"`
+	BlockSize   memory.Size `help:"size (in bytes) of encrypted blocks" default:"1KiB"`
+	DataType    int         `help:"Type of encryption to use for content and metadata (1=AES-GCM, 2=SecretBox)" default:"1"`
+	PathType    int         `help:"Type of encryption to use for paths (0=Unencrypted, 1=AES-GCM, 2=SecretBox)" default:"1"`
+}
+
+// LoadKey returns the root key for encrypting the data which is stored in the path
+// indicated by KeyFilepath field.
+//
+// It returns an error if:
+//
+// * KeyFilepath field is an empty string. Err value will be ErrKeyFilepathEmpty
+// * The file doesn't exist.
+// * There is an I/O error.
+//
+// TODO: if/v3-1541 refactor cache the key after the firs read. The problem is
+// with pkg/cfgstruct which is inflexible on accepting unexported keys nor
+// several different types as a type of storj.Key, [32]byte, etc.
+func (cfg *EncryptionConfig) LoadKey() (key storj.Key, err error) {
```

Comment:
The return value should be a pointer, i.e. `*storj.Key` instead of `storj.Key`.

`storj.Key` is a 32-byte array, not a slice. We want to reference it always via a pointer to avoid many copies of the encryption key in the memory as it is sensitive data.

Reply:
I had that question in the closed PR https://github.com/storj/storj/pull/1782/files/a77bb67c07b6f0dbde7d88329939659dbb3549f1#discussion_r278839646

I was waiting for the reply, however I forgot to moved to this PR.

Reply:
Only pointers will be use in the new implementation.